Tp1 239 cleanup package.json

dlopezvsr opened the pull request:
# Description

This PR cleans up the `package.json` npm scripts, removing redunant scripts already created on `tasks.py`, and also deleting duplicates. Some scripts where also migrated from `package.json` and to `tasks.py` in order to keep consistency on the scripts and leave as simple as possible the npm scripts section. Only scripts necessary to build the application were left, and also the main ones that are called from `tasks.py`.

Link to sample test page: https://foundation-s-tp1-239-cl-2nnvbk.herokuapp.com
Related PRs/issues: https://github.com/MozillaFoundation/foundation.mozilla.org/issues/11767

# Checklist

<!-- Check off items with `[x]` or cross out items that don't apply with `~~The description~~` -->

**Tests**
- [ ] Is the code I'm adding covered by tests?

**Changes in Models:**
- [ ] Did I update or add new fake data?
- [ ] Did I squash my migration?
- [ ] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [ ] Is my code documented?
- [ ] Did I update the READMEs or wagtail documentation?

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-1265)

danielfmiranda reviewed (commented):
Thanks for taking on this work @dlopezvsr!

The PR looks good to me, going to tag @mmmavis to take a look at package/npm changes as they are a little more familiar 👍

dlopezvsr commented:
@danielfmiranda thank you for your review. I'm also looking forward to @mmmavis comments. 🙌

mmmavis commented:
Sorry for the wait 😅 . I will give this PR a look tomorrow!

mmmavis commented on tasks.py:
```diff
@@ -595,3 +607,99 @@ def staging_db_to_review_app(ctx, review_app_name):
+@task
+def prettier_scss(ctx):
+    """Runs Prettier to format SCSS and CSS files."""
+    ctx.run('prettier "source/sass/**/*.scss" "source/js/**/*.scss" "network-api/networkapi/{,!(frontend)/**/}*.css"')
```
Similar output as running `inv prettier-scss`

mmmavis reviewed (changes requested):
Hey @dlopezvsr , thank you for diving into `package.json` and Docker to make things more streamlined! 


Apologies for leaving tons of comments. I'm not a Docker expert, so the best I could do was by testing each command. Unfortunately, I was getting errors for the majority of new inv commands introduced in this PR. A few were due to `run-s` and `run-p`, and some were due to "commands not found" in the Docker environments. I think those packages probably need to be installed globally or somehow configured for Docker to be properly run as `inv` command. Or we can leave them in package.json for simplicity. 

Let me know if I can help with anything. Maybe we can explore the Docker wonderland together to find solutions to these issues lol

mmmavis commented on tasks.py:
```diff
@@ -595,3 +607,99 @@ def staging_db_to_review_app(ctx, review_app_name):
+
+
+@task
+def prettier_js(ctx):
```
Note: I have `prettier` installed globally on my machine, so that is why I was able to run this command without hitting the `command not found` error

dlopezvsr commented:
> Hey @dlopezvsr , thank you for diving into `package.json` and Docker to make things more streamlined!
> 
> Apologies for leaving tons of comments. I'm not a Docker expert, so the best I could do was by testing each command. Unfortunately, I was getting errors for the majority of new inv commands introduced in this PR. A few were due to `run-s` and `run-p`, and some were due to "commands not found" in the Docker environments. I think those packages probably need to be installed globally or somehow configured for Docker to be properly run as `inv` command. Or we can leave them in package.json for simplicity.
> 
> Let me know if I can help with anything. Maybe we can explore the Docker wonderland together to find solutions to these issues lol

@mmmavis thank you so much for taking this deep look at the changes amd for all the comments, I will review each one and apply the necessary fixes. I remember to test most of them and run them without error, but I also made some changes later with some commands that I put back into the package.json due to some dependency errors in the build, so I may have missed removing them form `inv` commands.

dlopezvsr commented:
@mmmavis thank you so much again for all the comments and the exhaustive review. After taking a deeper look into the CI workflow, the scripts I had created, and based on all your comments, I decided that, as your mentioned, it makes totally more sense to keep npm-related scripts under `package.json` for several some important reasons. 

- The error related to `npm-s` and `run-p` is not encountered on the scripts because npm resolves them inside `node_modules/.bin` so at the moment we run the commands on the container, these are not found. Even we can install  `npm-s` and `run-p` globally, or use `npx `, keeping npm-related scripts in the `package.json` would be simpler and cleanner.
- Most of the scripts I had initially moved into `tasks.py` were added back into `package.json` due to other dependent scripts.
- Almost all the scripts are part of the CI pipeline and referenced to each other, on build and tests stages, also as part of the `continous-integration.yml` file, so as you highlighted on some scripts I missed to add back after removal, most of them need to be kept in `package.json`.

# Final version and changes
- As per Jhonatan [issue](https://github.com/MozillaFoundation/foundation.mozilla.org/issues/11767), I made sure playwright and other dependencies are installed in the dev Docker build, and confirmed `copy-db.js`  command is on `tasks.py`.
- I removed `docker:up` and `docker:down` as duplicated commands.
- I removed the scripts I initially created on tasks.py, and added back to `package.json` the npm-related commands.

@mmmavis let me know your comments :)

mmmavis reviewed (approved):
@dlopezvsr Thank you for walking me through the changes step by step!
